content/frameworks/react/virtualize-long-lists-react-window/index.md

```diff
@@ -0,0 +1,286 @@
+### FixedSizeList
+
+Use the **`FixedSizeList`** component if you have a long, one-dimensional list of equally sized items.
+
+```
+import React from 'react';
+import { FixedSizeList } from 'react-window';
+
+const items = [...] // some list of items
+ 
+const Row = ({ index, style }) => (
+  <div style={style}>
+     {/* define the row component using items[index) */}
+  </div>
+);
+ 
+const ListComponent = () => (
+  <FixedSizeList
+    height={500}
+    width={500}
+    itemSize={120}
+    itemCount={items.length}
+  >
+    {Row}
+  </FixedSizeList>
+);
+
+export default ListComponent;
+```
+
+* The `FixedSizeList` component accepts a `height`, `width` and `itemSize` prop to control the size of the items within the list.
+* A function that renders the rows is passed as a child to `FixedSizeList`. Details about the particular item can be accessed with the `index` argument (`items[index]`). 
+* A `style` parameter is also passed in to the row rendering method that **must** be attached to the row element. List items are absolutely positioned with their height and width values assigned inline, and the `style` parameter is responsible for this. 
+
+<div class="aside caution">
+Do not assign <code>height</code> and <code>width</code> properties to the list or the list item with an external CSS file. This would be ignored due to the fact that these style attributes are assigned inline.
+</div>
+
+The Glitch example shown earlier in this article shows an example of a `FixedSizeList` component.
+
+### VariableSizeList
+
+The `VariableSizeList` component works in the same way as a fixed size list, but instead expects a function for the `itemSize` prop instead of a specific value. Use this component instead if you need to render a list of items that have different sizes.
+
+```
+import React from 'react';
+import { VariableSizeList } from 'react-window';
+
+const items = [...] // some list of items
+ 
+const Row = ({ index, style }) => (
+  <div style={style}>
+     {/* define the row component using items[index) */}
+  </div>
+);
+
+const getItemSize = index => {
+  // return a size for items[index]
+}
+ 
+const ListComponent = () => (
+  <VariableSizeList
+    height={500}
+    width={500}
+    itemCount={items.length}
+    itemSize={getItemSize}
+  >
+    {Row}
+  </VariableSizeList>
+);
+
+export default ListComponent;
+```
+
+The following embed shows an example of this component.
+
+<div class="glitch-embed-wrap" style="height: 750px; width: 100%;">
+  <iframe
+    allow="geolocation; microphone; camera; midi; vr; encrypted-media"
+    src="https://glitch.com/embed/#!/embed/react-window-variable-list?path=src/ListComponent.js&previewSize=100"
+    alt="react-window-variable-list on Glitch"
+    style="height: 100%; width: 100%; border: 0;">
+  </iframe>
+</div>
+
+The item size function passed to `itemSize` prop randomizes the row heights in this example. In a real application however, there should be actual logic defining the sizes of each item.
```

Comment:
Might be worth trying to clarify how that logic comes to be? My wording is bad but something like:

```suggestion
The item size function passed to `itemSize` prop randomizes the row heights in this example. In a real application however, there should be actual logic defining the sizes of each item. Ideally, item sizes can be calculated based on data or obtained from an API.
```

Reply:
Updated 👍